### ISSUE TITLE
Penthouse dependancy is outdated

### ISSUE THREAD

**Issue by maximilianmessing** [mudbugmedia/critical-path-css-rails#5]
Penthouse is approaching its stable release. Maybe it is time to update the dependancies of this gem :) .

**Comment by michael-misshore** [comment]
@maximilianmessing, Sorry for the late reply as i'm just getting back in town. I'll write up a reply as soon as I can that explains why updating penthouse may not be possible.

**Comment by michael-misshore** [comment]
@maximilianmessing Currently, this gem is leveraging the command line option of Penthouse. Here is the section that states that the command line option is being supported for newer versions: https://github.com/pocketjoso/penthouse/blob/master/README.md#from-command-line

When this gem was created, I wasn't able to figure out a way to get the Node option working. Could you propose another solution that would allow us to stay up to date with Penthouse?

**Comment by milgner** [comment]
It should be possible to invoke the penthouse JS from within Ruby using [therubyracer](https://github.com/cowboyd/therubyracer) by specifying a Ruby function as the penthouse callback to retrieve the generated CSS. If you're interested I can give it a try tomorrow evening.

**Comment by michael-misshore** [comment]
@milgner, Any luck on getting this working using `therubyracer`?

**Comment by glebm** [comment]
Hey y'all, check out the approach in #18

**Comment by glebm** [comment]
On a related note, I have published a gem that wraps [critical]:

https://github.com/glebm/critical-rubygem

[critical]: https://github.com/glebm/critical-rubygem

**Comment by michael-misshore** [comment]
@glebm Sorry for the late reply. When I first saw your pull request come in, I got super excited since this has been something on the TODO list for awhile! I took a run through the changes and everything looks good. I'm trying to get some time to be able to play around and test these changes. I'll attempt to keep you up to date with the progress, and I may have to reach out if I have any questions relating your pull request.

Thank you so much for this contribution.
